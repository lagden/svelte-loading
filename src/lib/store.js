import {writable} from 'svelte/store'

const loadingMap = new Map()
const base = {
	open: false,
	isloading: false,
}

function loadingStore() {
	const {set, subscribe, update} = writable({...base})
	return {
		set,
		subscribe,
		open: () => update(n => {
			n.open = true
			n.isloading = true
			return n
		}),
		close: () => update(n => {
			n.open = false
			n.isloading = false
			return n
		}),
	}
}

export function unique(name) {
	if (loadingMap.has(name)) {
		return loadingMap.get(name)
	}
	loadingMap.set(name, loadingStore())
	return loadingMap.get(name)
}

const KEY = Symbol.for('store.loading')
const singleton = {}
singleton[KEY] = loadingStore()

export default singleton[KEY]
