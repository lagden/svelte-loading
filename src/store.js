import {writable} from 'svelte/store'

export const isloading = writable(false)
