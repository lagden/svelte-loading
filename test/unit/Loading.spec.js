import { setTimeout } from 'timers/promises'
import { get } from 'svelte/store'
import { tick } from 'svelte'
import {
	test,
	expect,
	beforeEach,
	vi,
} from 'vitest'
import {
	Loading,
	acts,
} from '$lib/index.js'

beforeEach(() => {
	document.body.innerHTML = ''
})

function doc_query(selector) {
	const node = document.querySelector(selector)
	if (!node) {
		throw new Error(`No element found for selector: ${selector}`)
	}
	return node
}

test('Loading', async () => {
	const target = document.body
	new Loading({ target })
	acts.show(true)

	await setTimeout(1500)
	const div = doc_query('div._tadashi_svelte_loading')
	expect(div).toMatchSnapshot()
	acts.show(false)
})

test('Loading name', async () => {
	const target = document.body
	const name = 'Jelly'
	new Loading({ target, props: {
		name,
		animation: name,
	}})
	acts.show(true, name)

	await setTimeout(1500)
	const div = doc_query('div._tadashi_svelte_loading')
	expect(div).toMatchSnapshot()
})

test('Loading not exist', async () => {
	const target = document.body
	const name = 'Uebaaa'
	acts.show(true, name)
	new Loading({ target, props: {
		name,
		animation: name,
	}})

	await setTimeout(1500)
	const div = doc_query('div._tadashi_svelte_loading')
	expect(div).toMatchSnapshot()
})
