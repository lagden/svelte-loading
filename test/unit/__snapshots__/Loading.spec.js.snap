// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loading 1`] = `
<div
  class="_tadashi_svelte_loading s-rumqwqG3Ogas"
  style=""
>
  <svg
    class="ring s-VzEkOmBnbMsD"
    stroke-width="5"
    viewBox="25 25 50 50"
  >
    <circle
      class="s-VzEkOmBnbMsD"
      cx="50"
      cy="50"
      r="20"
    />
  </svg>
  <!--&lt;Ring&gt;-->
</div>
`;

exports[`Loading name 1`] = `
<div
  class="_tadashi_svelte_loading s-rumqwqG3Ogas"
  style=""
>
  <div
    class="jelly s-w6j_BusLbHkw"
  />
   
  <svg
    class="jelly-maker s-w6j_BusLbHkw"
    height="0"
    width="0"
  >
    <defs
      class="s-w6j_BusLbHkw"
    >
      <filter
        class="s-w6j_BusLbHkw"
        id="uib-jelly-ooze"
      >
        <fegaussianblur
          class="s-w6j_BusLbHkw"
          in="SourceGraphic"
          result="blur"
          stdDeviation="6.25"
        />
        <fecolormatrix
          class="s-w6j_BusLbHkw"
          in="blur"
          mode="matrix"
          result="ooze"
          values="1 0 0 0 0  0 1 0 0 0  0 0 1 0 0  0 0 0 18 -7"
        />
        <feblend
          class="s-w6j_BusLbHkw"
          in="SourceGraphic"
          in2="ooze"
        />
      </filter>
    </defs>
  </svg>
  <!--&lt;Jelly&gt;-->
</div>
`;

exports[`Loading not exist 1`] = `
<div
  class="_tadashi_svelte_loading s-rumqwqG3Ogas"
  style=""
>
  <svg
    class="ring s-VzEkOmBnbMsD"
    stroke-width="5"
    viewBox="25 25 50 50"
  >
    <circle
      class="s-VzEkOmBnbMsD"
      cx="50"
      cy="50"
      r="20"
    />
  </svg>
  <!--&lt;Ring&gt;-->
</div>
`;
